streaming/src/data_reader.cc

```diff
@@ -286,37 +437,42 @@ void DataReader::NotifyConsumed(std::shared_ptr<DataBundle> &message) {
-    NotifyConsumedItem(channel_info, message->seq_id);
+  if (queue_info.target_message_id <= message->meta->GetLastMessageId()) {
+    NotifyConsumedItem(channel_info, message->meta->GetLastMessageId());
 
     channel_map_[channel_info.channel_id]->RefreshChannelInfo();
-    if (queue_info.last_seq_id != QUEUE_INVALID_SEQ_ID) {
-      uint64_t original_target_seq_id = queue_info.target_seq_id;
-      queue_info.target_seq_id = std::min(
-          queue_info.last_seq_id,
-          message->seq_id + runtime_context_->GetConfig().GetReaderConsumedStep());
+    if (queue_info.last_message_id != QUEUE_INVALID_SEQ_ID) {
+      uint64_t original_target_message_id = queue_info.target_message_id;
+      queue_info.target_message_id =
+          std::min(queue_info.last_message_id,
+                   message->meta->GetLastMessageId() +
+                       runtime_context_->GetConfig().GetReaderConsumedStep());
       channel_info.last_queue_target_diff =
-          queue_info.target_seq_id - original_target_seq_id;
+          queue_info.target_message_id - original_target_message_id;
     } else {
       STREAMING_LOG(WARNING) << "[Reader] [QueueInfo] channel id " << message->from
-                             << ", last seq id " << queue_info.last_seq_id;
+                             << ", last message id " << queue_info.last_message_id;
     }
     STREAMING_LOG(DEBUG) << "[Reader] [Consumed] Trigger notify consumed"
-                         << ", channel id => " << message->from << ", last seq id => "
-                         << queue_info.last_seq_id << ", target seq id => "
-                         << queue_info.target_seq_id << ", consumed seq id => "
-                         << message->seq_id << ", last message id => "
-                         << message->meta->GetLastMessageId() << ", bundle type => "
+                         << ", channel id=" << message->from
+                         << ", last message id=" << queue_info.last_message_id
+                         << ", target message id=" << queue_info.target_message_id
+                         << ", consumed message id=" << message->meta->GetLastMessageId()
+                         << ", bundle type="
                          << static_cast<uint32_t>(message->meta->GetBundleType())
-                         << ", last message bundle ts => "
+                         << ", last message bundle ts="
                          << message->meta->GetMessageBundleTs();
   }
 }
 
 bool StreamingReaderMsgPtrComparator::operator()(const std::shared_ptr<DataBundle> &a,
                                                  const std::shared_ptr<DataBundle> &b) {
+  if (comp_strategy == ReliabilityLevel::EXACTLY_ONCE) {
+    if (a->last_barrier_id != b->last_barrier_id)
+      return a->last_barrier_id > b->last_barrier_id;
+  }
   STREAMING_CHECK(a->meta);
-  // We use hash value of id for stability of message in sorting.
+  // we proposed push id for stability of message in sorting
```

Comment:
// We proposed fixed id sequnce for stability of message in sorting.

Reply:
done.